### PR TITLE
Ukrainian language

### PR DESCRIPTION
**Pull request by sleepyFlowey**
Ukrainian language

### CODE REVIEW

File: translations/uk_UA.json [hunk 1]
```diff
@@ -0,0 +1,667 @@
...
+  "REQUIRES_CLEARING_IMAGE_CACHE_AND_RE_INDEXING": "Необхідно очистити кеш зображень та повторне переіндексування",
+  "RESCAN_VIDEOS": "Повторне сканування відео",
+  "RESET_BRIGHTNESS": "Скидання яскравості",
+  "RESET_TO_DEFAULT": "Встановити на",
+  "RESPECT_NO_MEDIA": "Дотримуватися .nomedia",
+  "RESPECT_NO_MEDIA_SUBTITLE": "Виключає теки, що містять .nomedia",
+  "RESTART": "Перезавантажити",
+  "RESTORED_BACKUP_SUCCESSFULLY": "Резервна копія відновлення",
+  "RESTORED_BACKUP_SUCCESSFULLY_SUB": "Файл резервної копії успішно відновлено",
+  "RESTORE_BACKUP": "Відновити резервну копію,
```
Comment (VerySweetBread) [inline translations/uk_UA.json:501]: You missed `"`

File: translations/uk_UA.json [hunk 2]
```diff
@@ -0,0 +1,667 @@
...
+  "DAY": "День",
+  "DAYS": "Дня",
+  "DEFAULT": "За замовчуванням",
+  "DEFAULT_BACKUP_LOCATION": "Розташування Резервних копій за Замовчуванням",
+  "DEFAULT_COLOR": "Тема за Замовчуванням",
+  "DEFAULT_COLOR_SUBTITLE": "Встановлення кольору, що буде використовуватись плеєром",
+  "DEFAULT_DOWNLOAD_LOCATION": "Папка для завантаження за Замовчуванням",
+  "DEFAULT_LIBRARY_TAB": "Бібліотека за замовуванням",
+  "DELETE": "Видалити",
+  "DELETE_FILE_CACHE_SUBTITLE": "Delete _FILES_COUNT_ file representing _TOTAL_SIZE_?",
```
Comment (VerySweetBread) [inline translations/uk_UA.json:133]: Forgot to translate

File: translations/uk_UA.json [hunk 3]
```diff
@@ -498,7 +498,7 @@
   "RESTART": "Перезавантажити",
   "RESTORED_BACKUP_SUCCESSFULLY": "Резервна копія відновлення",
   "RESTORED_BACKUP_SUCCESSFULLY_SUB": "Файл резервної копії успішно відновлено",
-  "RESTORE_BACKUP": "Відновити резервну копію,
+  "RESTORE_BACKUP": "Відновити резервну копію,"
```
Comment (VerySweetBread) [inline translations/uk_UA.json:501]: XD Don't hurry, put `"` *before* a comma

**Comment by VerySweetBread** [comment]
You can check your translation puting .json file in language chooser. (Your current file causes an error
Screenshot_20240406_001618
)

**Comment by sleepyFlowey** [comment]
Used the localization file in the app, no errors.

Screenshot_20240406-002157_Namida

**Comment by VerySweetBread** [comment]
hM
SmartSelect_20240406_002913
Anyway, it works

File: langs.json
Comment (VerySweetBread) [inline langs.json:1]: Something wrong with whitespaces... Will be the mainterner problem)

**Review by VerySweetBread — APPROVED** [review]

**Comment by sleepyFlowey** [comment]
The Ukrainian language has some letters that are in extended Cyrillic fonts, "І" and "Ї". So, it's a font problem.

**Comment by VerySweetBread** [comment]
> The Ukrainian language has some letters that are in extended Cyrillic fonts, "І" and "Ї". So, it's a font problem.

Yep, and it's strange, 'cuz they are same Cyrillic letters as `ы` and `й`

**Comment by sleepyFlowey** [comment]
> Yep, and it's strange, 'cuz they are same Cyrillic letters as `ы` and `й`
 
 And the Belarusian language has the letter "Ў", it's just that these languages are less popular than Russian, and so you don't see them in fonts as often. In this case, I don't think it's a big problem, because if the devices have support for the Ukrainian language by default, then everything will be fine in the app.

**Comment by MSOB7YY** [comment]
woahh thank u soooo muchhhhh, gonna merge now

**Comment by Andrew1397** [comment]
> hM SmartSelect_20240406_002913 Anyway, it works

How can this problem be solved?
My device supports the Ukrainian language and all programs perfectly display all letters, but in "Namida" there is such a problem

**Comment by MSOB7YY** [comment]
@Andrew1397 yeah font problem, tracking at https://github.com/namidaco/namida/issues/183

**Comment by Andrew1397** [comment]
And with which font are there no problems?
Because I tried all the fonts available in Crdroid, but the problem remains. Or I did not understand at all what fonts you are talking about

**Comment by MSOB7YY** [comment]
@Andrew1397 the issue is with the custom font that namida uses, not your problem. i will check what i can do with that

**Comment by Andrew1397** [comment]
Now I understand.
Thank you.

**Pull request merged**